### ISSUE TITLE
`unnecessary_safety_comment` does not lint for "orphan" comments

### ISSUE THREAD

**Issue by ojeda** [rust-lang/rust-clippy#14556]
### Summary

`unnecessary_safety_comment` does not lint for "orphan" comments, i.e. the last (or only) comment.

Possibly related: https://github.com/rust-lang/rust-clippy/issues/14553.
Possibly related: https://github.com/rust-lang/rust-clippy/issues/14554.
Possibly related: https://github.com/rust-lang/rust-clippy/issues/14555.

### Lint Name

`unnecessary_safety_comment`

### Reproducer

https://godbolt.org/z/GrjxKc9nW

I tried this code:

```rust
// This line should avoid #14553.

mod x {}

// SAFETY: ...
```

I expected to see this happen: lint.

Instead, this happened: no lint.

Same if it is a single comment:

```rust
// This line should avoid #14553.

// SAFETY: ...
```

### Version

```text
rustc 1.86.0 (05f9846f8 2025-03-31)
rustc 1.88.0-nightly (17ffbc81a 2025-04-04)
```
